src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -4274,16 +4278,26 @@ internal static void DisposeOnShutdown()
 #endif
         {
             Debug.Assert(EventSource.IsSupported);
-
+            List<EventSource> sourcesToDispose = new List<EventSource>();
```

Comment:
Is it worth only instantiating this if we find a source to dispose?  Or is that pretty much guaranteed to always be the case?

Reply:
Unless we make changes elsewhere I expect there will always be EventSources in this list.